src/lib.rs

```diff
@@ -196,32 +194,38 @@ impl SystemRuntime {
         tokio::spawn(connection);
 
         trace!("configure {}", LOOPBACK_DEV);
-        if let Ok(ipv6) = "::1/128".parse::<Ipv6Network>(){
-            if let Err(e) = add_address_ipv6(LOOPBACK_DEV, ipv6, handle.clone()).await{
-                error!("{}",e);
+        if let Ok(ipv6) = "::1/128".parse::<Ipv6Network>() {
+            if let Err(e) =
+                add_address_ipv6(LOOPBACK_DEV, ipv6, handle.clone()).await
+            {
+                error!("{}", e);
             };
         };
-        if let Ok(ipv4) = "127.0.0.1/8".parse::<Ipv4Network>(){
-            if let Err(e) = add_address_ipv4(LOOPBACK_DEV, ipv4, handle.clone()).await{
-                error!("{}",e);
+        if let Ok(ipv4) = "127.0.0.1/8".parse::<Ipv4Network>() {
+            if let Err(e) =
+                add_address_ipv4(LOOPBACK_DEV, ipv4, handle.clone()).await
+            {
+                error!("{}", e);
             }
         };
 
-        if let Err(e) = set_link_up(handle.clone(), LOOPBACK_DEV).await{
-            error!("{}",e);
+        if let Err(e) = set_link_up(handle.clone(), LOOPBACK_DEV).await {
+            error!("{}", e);
         }
 
         trace!("configure eth0");
-        if let Ok(ipv6) = "fe80::2/64".parse::<Ipv6Network>(){
-            if let Err(e) = add_address_ipv6("eth0", ipv6, handle.clone()).await{
-                error!("{}",e);
+        if let Ok(ipv6) = "fe80::2/64".parse::<Ipv6Network>() {
+            if let Err(e) = add_address_ipv6("eth0", ipv6, handle.clone()).await
+            {
+                error!("{}", e);
             }
         };
-        if let Err(e) = set_link_up(handle.clone(), "eth0").await{
-            error!("{}",e);
+        if let Err(e) = set_link_up(handle.clone(), "eth0").await {
+            error!("{}", e);
         }
 
         show_network_info(handle).await;
+        spawn_power_button_listener();
```

Comment:
Yes, makes sense. Placed it in a new function, and used the comment blocks. I think this will likely move again, but until then we have a dedicated place.